src/index.ts

```diff
@@ -72,7 +73,13 @@ export { noopCacheProvider, IServiceProviderConfig, SamlConfig };
 const withSpidAuthMiddleware = (
   acs: AssertionConsumerServiceT,
   clientLoginRedirectionUrl: string,
-  clientErrorRedirectionUrl: string
+  clientErrorRedirectionUrl: string,
+  logCallback?: (
+    sourceIp: string | null,
+    payload: string,
+    timestamp: string,
+    isRequest: boolean
```

Comment:
this should be a literal with type
`type CALLBACK_PAYLOAD_TYPE = "REQUEST" | "RESPONSE"`

avoid using booleans as parameters, they do not scale and makes calls obscure